sys/kern/vm_physmem.c

```diff
@@ -293,7 +293,6 @@ static void pm_free_from_seg(vm_physseg_t *seg, vm_page_t *page) {
   PAGECOUNT(page)++;
   page->flags |= PG_MANAGED;
   for (unsigned i = 0; i < page->size; i++) {
-    pmap_page_remove(&page[i]);
```

Comment:
I think it would be a good idea to add an assertion to `vm_page_free()` which checks that there are no mappings of that page, i.e. that `page->pv_list` is empty.

Reply:
I agree it would be nice to check here that page is not referenced anymore. If I used `&page[i].pv_list` that would be half-correct since a page can be mapped by `pmap_kenter`, which does not maintain `pv_entry` list. Unfortunately we don't maintain reference count on a page. If we did, I'd definitely check it here.

I added a comment that the check is missing here. Is that ok?

Reply:
Oh... one more thing... If I checked `&page[i].pv_list` here I would probably need to lock `pv_list_lock` which is exactly what I'm trying to avoid :)

Reply:
IIUC, there must be no mappings of a page in order to call `vm_page_free()` on it. Furthermore, no one else will be allocated the page until `pm_free_from_seg()` does its thing, so at the beginning of `vm_page_free()` we can be sure that we're the only users of the page. I think it's safe to check `pv_list` without acquiring `pv_list_lock`.

Reply:
Ok. It's still a useful temporary solution capable of spotting an error :)